src/Microsoft.OData.Client/ALinq/Evaluator.cs

```diff
@@ -178,6 +178,17 @@ internal override Expression VisitMethodCall(MethodCallExpression m)
                 {
                     this.cannotBeEvaluated = true;
                 }
+
+                if (m.Method.DeclaringType != null && (m.Method.DeclaringType.ContainsGenericParameters || m.Method.DeclaringType.IsByRefLike))

```

Comment:
I don't particularly understand why evaluation of the condition `m.Method.DeclaringType.ContainsGenericParameters` to `true` results into us deciding that `this.cannotBeEvaluated = true`